packages/react-reconciler/src/ReactFiber.js

```diff
@@ -270,11 +270,6 @@ export function createWorkInProgress(
     // We already have an alternate.
     // Reset the effect tag.
     workInProgress.effectTag = NoEffect;
-
-    // The effect list is no longer valid.
-    workInProgress.nextEffect = null;
-    workInProgress.firstEffect = null;
-    workInProgress.lastEffect = null;
   }
```

Comment:
Is it important to preserve them now? Why?

Reply:
https://github.com/facebook/react/pull/11901#discussion_r159291448